subsys/bluetooth/audio/ascs.c

```diff
@@ -1884,13 +1884,25 @@ static int ase_metadata(struct bt_ascs_ase *ase, uint8_t op,
 	}
 
 	if (err) {
+		uint8_t reason = 0x00;
+
+		/* Set reason to the value of the Metadata Type */
+		if (err > 0) {
+			reason = err;
+			err = BT_ASCS_RSP_METADATA_REJECTED;
+		} else {
+			err = BT_ASCS_RSP_METADATA_INVALID;
+			if (buf->len) {
+				reason = *buf->data;
```

Comment:
What is in `*buf->data;`? 

Is there any way for us to send `BT_ASCS_RSP_METADATA_UNSUPPORTED` with this approach? Is there a test for that? 

As per the ASCS spec, the reason for `BT_ASCS_RSP_METADATA_REJECTED`, `BT_ASCS_RSP_METADATA_UNSUPPORTED` and `BT_ASCS_RSP_METADATA_INVALID` is the Metadata type, so why not handle them the same? 

Can we get away with only responding with `BT_ASCS_RSP_METADATA_REJECTED` if application rejects it? Do we need to send `BT_ASCS_RSP_METADATA_INVALID` to pass qualification?